compiler/damlc/tests/daml-test-files/ExceptionTryCatch.daml

```diff
@@ -0,0 +1,41 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @SINCE-LF 1.dev
+
+-- | Test the try-catch syntax.
+module ExceptionTryCatch where
+
+import DA.Exception
```

Comment:
maybe add a test that uses guards somewhere. Perhaps also worth testing that you get a more or less sensible error if you have multiple patterns in a case.